lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/InitMavenRequestTest.java

```diff
@@ -0,0 +1,89 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.lemminx.extensions.maven;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.io.File;
+import java.util.Properties;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(NoMavenCentralIndexExtension.class)
+public class InitMavenRequestTest {
+
+	private static final String TEST_LOCAL_REPO_PATH = "/.m2/bug383repository";
+	private static final String MAVEN_LOCAL_REPO_PROPERTY_NAME = "maven.repo.local";
+	
+	private static final String TEST_DUMMY_PROPERTY_NAME = "DUMMY_USER_HOME";
+	private static final String HOSTNAME_ENV_VARIABLE_NAME = "HOSTNAME";
+	private static final String MAVEN_PROJECTBASEDIR_ENV_VARIABLE_NAME = "MAVEN_PROJECTBASEDIR";
+	
+
+	@Test
+	public void testUserSettingsWithSystemProperty() throws Exception {
+		final File localSettingsFile = new File(getClass().getResource("/settings-local-repo-test-sys-property.xml").toURI());
+		String localRepoProperty = System.getProperty(MAVEN_LOCAL_REPO_PROPERTY_NAME);
+		Properties properties = System.getProperties();
+		try {
+			properties.remove(MAVEN_LOCAL_REPO_PROPERTY_NAME); // Not to use local repository set from properties
+			
+			// Add a DUMMY environment variable to check later if it's really resolved to a real path
+			// Avoiding any ${HOME} and ${user.home} variable/properties values as they are 
+			// used as defaults in maven
+			String hostname = "/" + System.getenv(HOSTNAME_ENV_VARIABLE_NAME);
+			properties.put(TEST_DUMMY_PROPERTY_NAME, hostname);
+			System.setProperties(properties);
+			
+			MavenLemminxExtension plugin = new MavenLemminxExtension();
+			plugin.settings.setGlobalSettings(null);
+			plugin.settings.setUserSettings(localSettingsFile.getAbsolutePath());
+			
+			File localRepositoryPath = plugin.getMavenSession().getRequest().getLocalRepositoryPath();
+			assertNotNull(localRepositoryPath);
+			assertEquals(hostname + TEST_LOCAL_REPO_PATH, localRepositoryPath.getAbsolutePath());
+		} finally {
+			// Restore original System properties
+			properties.remove(TEST_DUMMY_PROPERTY_NAME);
+			if (localRepoProperty != null) {
+				properties.put(MAVEN_LOCAL_REPO_PROPERTY_NAME, localRepoProperty);
+			}
+			System.setProperties(properties);
+		}
+	}
+	
+	@Test
+	public void testUserSettingsWithEnvVariables() throws Exception {
+		final File localSettingsFile = new File(getClass().getResource("/settings-local-repo-test-env-variable.xml").toURI());
+		String localRepoProperty = System.getProperty(MAVEN_LOCAL_REPO_PROPERTY_NAME);
+		Properties properties = System.getProperties();
+		try {
+			properties.remove(MAVEN_LOCAL_REPO_PROPERTY_NAME); // Not to use local repository set from properties
+			System.setProperties(properties);
```

Comment:
No System property has to/should be set if you want to test env-vars.

Reply:
If you mean `MAVEN_LOCAL_REPO_PROPERTY_NAME`:
```
private static final String MAVEN_LOCAL_REPO_PROPERTY_NAME = "maven.repo.local";
```
then we have to remove this system property, because otherwise it will be used to create a Maven Local Repo overridinf the values provided by configuration files.

Reply:
Ah yes, my bad.